src/fieri/app/models/metrics_runner.rb

```diff
@@ -24,4 +24,8 @@ def cookbook_api_response(params)
   def cookbook_version_api_response(params)
     SupermarketApiRunner.new.cookbook_version_api_response(params['cookbook_name'], params['cookbook_version'])
   end
+
+  def source_repo(cookbook_json)
+    cookbook_json.match(%r{(?<=github.com\/)[\w-]+\/[\w-]+}).to_s
```

Comment:
On your first point, I can see that, having something like a GithubMetric class that the Contributing Metric and Testing Metric both inherit from.

On your second point, I will check into doing that.